### PR TITLE
Document the need for COSI, and Developer guide

### PR DESCRIPTION
**Pull request by wlan0**
This is ready for review

https://github.com/kubernetes-sigs/container-object-storage-interface-spec/issues/14

### CODE REVIEW

**Comment by wlan0** [comment]
@xing-yang please review

**Review by tparikh — APPROVED** [review]
LGTM

File: README.md [hunk 1]
```diff
@@ -1,43 +1,69 @@
...
+```
 
-If you are new to a SIG Storage project, check out this [contributing guide](https://github.com/kubernetes/community/blob/master/sig-storage/CONTRIBUTING.md) 
-and the [community page](https://github.com/kubernetes/community/tree/master/sig-storage). 
+3. Clean and Build
 
-If you are new to the SIG Storage COSI project, check out the [spec](https://github.com/kubernetes-sigs/container-object-storage-interface-spec/blob/master/spec.md), [KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1979-object-storage-support), and [project board](https://github.com/orgs/kubernetes-sigs/projects/).
+```sh
+# cleans up old build files
+make clobber
```
Comment (brahmaroutu) [inline README.md:42]: why make clobber? can make clean do this?

File: README.md [hunk 2]
```diff
@@ -1,43 +1,69 @@
...
- [CSI Adapter](https://github.com/kubernetes-sigs/container-object-storage-interface-csi-adapter) <br /> 
- &emsp; - images: cosi-node-adapter 
+# Container Object Storage Interface Spec
 
-COSI deprecated repos:   
+This repository hosts the gRPC API for the Container Object Storage Interface (COSI) standard. The interfaces defined in the [gRPC specification](cosi.proto) are meant to be the common interface for object storage provisioning and management across various object storage vendors.
 
+For more information about the COSI effort, visit our [documentation](https://github.com/kubernetes-sigs/container-object-storage-interface-api/tree/master/docs/index.md).
 
+## Why another standard?
```
Comment (jeffvance) [inline README.md:9]: This section is a dup from another doc. Better to link to a single source here.

File: README.md (hunk 1 above)
Comment (brahmaroutu) [reply README.md:42]: Looks like this is in addition to `make clean` as there are some intermediate files that need to be cleaned up while keeping the generated protobuf

File: README.md [hunk 3]
```diff
@@ -1,43 +1,69 @@
...
+The existing primitives in CSI do not apply to objectstorage. Thus the need for a new standard to automate the management of objectstorage.
 
-# Container Object Storage Specification
+## Developer Guide
 
-Kubernetes specific Container Object Storage Interface (COSI) components.
+All API definitions **_MUST_** satisfy the following requirements:
 
-## Community, discussion, contribution, and support
+ - Must be backwards compatible
```
Comment (xing-yang) [inline README.md:24]: I'm not sure if this is practical for alpha APIs?  In Kubernetes, if API stays in alpha, it does not need to support backward compatibility; but once it reaches beta, it has to support backward compatibility moving forward.

File: README.md [hunk 4]
```diff
@@ -1,43 +1,69 @@
...
+ - Must be in-sync with the API definitions in [sigs.k8s.io/container-object-storage-interface-api](https://github.com/kubernetes-sigs/container-object-storage-interface-api)
```
Comment (xing-yang) [inline README.md:25]: COSI spec is a lower level API.  I think it should not talk about the higher level API.  It is the other way around.

Comment (wlan0) [reply README.md:25]: These two are closely inter-related.  we want to move them in lockstep.i.e. If a change is made here, then it should also be made in api.

File: README.md (hunk 3 above)
Comment (wlan0) [reply README.md:24]: This is a practical guideline for the community contributors who want to start contributing. I think between us in the team, we can have an understanding about it until we reach v1 status

**Comment by wlan0** [comment]
@xing-yang I've addressed your comments

File: README.md (hunk 3 above)
Comment (xing-yang) [reply README.md:24]: I think the language in the spec repo is supposed to be precise.  We can't require alpha APIs to be backward compatible.  Alpha means it is experimental.  If this requirement is added here but we are not really going to enforce it, then we should either loosen the requirement or remove it.

File: README.md (hunk 1 above)
Comment (wlan0) [reply README.md:42]: yeah

File: README.md (hunk 3 above)
Comment (wlan0) [reply README.md:24]: makes sense. Ive updated the docs

**Pull request merged**